src/dali/plugin/input/csv/binance.py

```diff
@@ -0,0 +1,219 @@
+                }
+
+                # in binance, you can pay fees with lots of different currencies. Since dali does not allow us to specify a fee currency, we use the normalized USD fee amount
+                # this impacts buy & sell transactions
+
+                # staking rewards & commission do not have any fees and only have a primary asset
+                if transaction_type in [_REFERRAL_COMMISSION, _STAKING_REWARDS]:
+                    granular_transaction_type = Keyword.INCOME if transaction_type == _REFERRAL_COMMISSION else Keyword.INTEREST
+                    currency: str = line[self.__CURRENCY_INDEX].strip()
+
+                    result.append(
+                        InTransaction(
+                            **(
+                                common_params
+                                | {
+                                    "asset": currency,
+                                    "exchange": self.__BINANCE,
+                                    "holder": self.account_holder,
+                                    "transaction_type": granular_transaction_type.value,
+                                    "spot_price": line[self.__PRIMARY_ASSET_SPOT_PRICE].strip(),
+                                    "crypto_in": line[self.__PRIMARY_ASSET_AMOUNT].strip(),
+                                    "fiat_fee": "0",
+                                }
+                            )
+                        )
+                    )
+                elif transaction_type == _CRYPTO_DEPOSIT:
+                    currency: str = line[self.__CURRENCY_INDEX].strip()
+
+                    result.append(
+                        IntraTransaction(
+                            **(
+                                common_params
+                                | {
+                                    "asset": currency,
+                                    "crypto_sent": "0",
+                                    "crypto_received": line[self.__PRIMARY_ASSET_AMOUNT].strip(),
+                                    "spot_price": line[self.__PRIMARY_ASSET_SPOT_PRICE].strip(),
+                                    # most likely, funds are coming from the user/tax payer, but we can't say for sure so we use unknown
+                                    # and let the user manually input the owner of these funds.
+                                    "from_exchange": Keyword.UNKNOWN.value,
+                                    "from_holder": Keyword.UNKNOWN.value,
+                                    "to_exchange": self.__BINANCE,
+                                    "to_holder": self.account_holder,
+                                }
+                            )
+                        )
+                    )
+                elif transaction_type == _CRYPTO_WITHDRAWAL:
+                    # TODO withdrawals can have fees in binance, but it doesn't look like IntraTransaction supports recording fees, not sure what to do here
```

Comment:
In intra transactions, the fee is implicit (it's crypto_received - crypto_sent). However for more scenarios (e.g. fee in a different currency, check https://github.com/eprbell/rp2/blob/main/docs/user_faq.md#how-to-represent-fiat-vs-crypto-transaction-fees)

Reply:
Got it! I think in my recent change I've handled this properly by creating an outtransaction for fees (since the currency could be a number of different assets)